snuba/datasets/__init__.py

```diff
@@ -25,8 +25,10 @@ class Dataset(object):
     def __init__(self,
             dataset_schemas: DatasetSchemas,
             *,
-            processor: MessageProcessor,
-            default_topic: str,
+            processor: Optional[MessageProcessor],
+            default_topic: Optional[str],
+            # PR 479 changes the architecture of the dataset
```

Comment:
All this change will automatically disappear when PR 479 gets committed and we will have  fewer nullable fields out of datasets.